gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/PublisherApi.java

```diff
@@ -0,0 +1,498 @@
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * EDITING INSTRUCTIONS
+ * This file was generated from the file google/pubsub/v1/pubsub.proto,
+ * and updates to that file get reflected here through a regular refresh process.
+ * However, manual additions are allowed because the refresh process performs
+ * a 3-way merge in order to preserve those manual additions. In order to not
+ * break the refresh process, only certain types of modifications are
+ * allowed.
+ *
+ * Allowed modifications - currently there is only one type allowed:
+ * 1. New methods (these should be added to the end of the class)
+ *
+ * Happy editing!
+ */
+package com.google.gcloud.pubsub.spi;
+
+import com.google.protobuf.Empty;
+import com.google.pubsub.v1.DeleteTopicRequest;
+import com.google.pubsub.v1.GetTopicRequest;
+import com.google.pubsub.v1.ListTopicSubscriptionsRequest;
+import com.google.pubsub.v1.ListTopicSubscriptionsResponse;
+import com.google.pubsub.v1.ListTopicsRequest;
+import com.google.pubsub.v1.ListTopicsResponse;
+import com.google.pubsub.v1.PublishRequest;
+import com.google.pubsub.v1.PublishResponse;
+import com.google.pubsub.v1.PublisherGrpc;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.Topic;
+
+import io.gapi.gax.grpc.ApiCallable;
+import io.gapi.gax.grpc.PageDescriptor;
+import io.gapi.gax.grpc.ServiceApiSettings;
+import io.gapi.gax.internal.ApiUtils;
+import io.gapi.gax.protobuf.PathTemplate;
+import io.grpc.Channel;
+import io.grpc.ManagedChannel;
+
+import java.io.IOException;
+import java.util.List;
+
+// Manually-added imports: add custom (non-generated) imports after this point.
+
+
+
+// AUTO-GENERATED DOCUMENTATION AND SERVICE - see instructions at the top of the file for editing.
+/**
+ * The service that an application uses to manipulate topics, and to send
+ * messages to a topic.
+ */
+@javax.annotation.Generated("by the veneer generator")
+public class PublisherApi implements AutoCloseable {
```

Comment:
* I have thought about having a functional interface, but that becomes a little complex for these classes. The reason is that every api call ends up generating 3 methods in the class - a flattened form, a form that takes a request object, and a form that returns an ApiCallable, and each one I listed forwards to the next one. If someone wanted to implement this interface for testing, this foists more work on them than is ideal (they would just repeat most of the forwarding code). One workaround for this is to add multiple layers in, one layer that contains the syntactic sugar methods (flattened form, request form) which forward to the callable form, and another layer that returns a mocked callable somehow. But that is a bit of complexity to add, and anyway, mocking out the channel actually seems like less work. 
* service split: Yes, this is derived from the proto. This makes sense because they would be used in different contexts, likely mutually exclusive. The only time both would be used is if someone writes a component that subscribes to one topic, processes the messages, then publishes to another topic. 
* GetIamPolicy: I don't know about the history of this; the functionality seems to be defined in its own proto now: https://github.com/google/googleapis/blob/master/google/iam/v1/iam_policy.proto